### PR TITLE
Fix fetching past events

### PR DESCRIPTION
**Pull request by r-czajkowski**
If we want to get past events eg. by all operators, and pass an empty array to indexed param eg. `{filter: { operator: [] }}` to the [`getPastEvent`](https://web3js.readthedocs.io/en/v1.2.0/web3-eth-contract.html#getpastevents), all emitted events will be returned. To avoid this, before calling `getPastEvent` with an empty array for the indexed param, we check that the filter param is not an empty array and then call ` getPastEvent` function.

### CODE REVIEW

File: solidity/dashboard/src/services/tbtc-rewards.service.js [hunk 1]
```diff
@@ -25,13 +26,16 @@ const fetchTBTCRewards = async (web3Context, beneficiaryAddress) => {
     transferEventSearchFilter
   )
 
+  const depositCreatedFilterParam = isEmptyArray(transferEventToBeneficiary)
```
Comment (pdyraga) [inline solidity/dashboard/src/services/tbtc-rewards.service.js:29]: If my understanding is correct, I think the name of this `const` is a little misleading as we are not talking about created deposits but rather about deposits that have been successfully closed. This is out of scope of this PR though, so let's leave it as-is here.

File: solidity/dashboard/src/services/token-staking.service.js [hunk 2]
```diff
@@ -218,12 +219,15 @@ export const getOperatorsOfOwner = async (owner) => {
...
-    })
-  ).reduce(reduceByOperator, {})
+  let transferEventsByOperators = {}
+  if (!isEmptyArray(operators)) {
+    transferEventsByOperators = (
+      await stakingContract.getPastEvents("StakeOwnershipTransferred", {
+        fromBlock: CONTRACT_DEPLOY_BLOCK_NUMBER.stakingContract,
+        filter: { operator: operators },
+      })
+    ).reduce(reduceByOperator, {})
```
Comment (pdyraga) [inline solidity/dashboard/src/services/token-staking.service.js:229]: I need help to understand it. Say we have `StakeOwnershipTransferred(operator = X, newOwner = A)` and then `StakeOwnershipTransferred(operator = X, newOwner = B)`. What we are doing here is that we are eliminating `newOwner = A` event, correct?

**Comment by pdyraga** [comment]
Heads up @nkuba, we will have to backport this change to https://github.com/keep-network/keep-core/pull/1989 you are reviewing.

**Comment by pdyraga** [comment]
Tested with multiple grants and delegations, works as advertised.

File: solidity/dashboard/src/services/tbtc-rewards.service.js (hunk 1 above)
Comment (r-czajkowski) [reply solidity/dashboard/src/services/tbtc-rewards.service.js:29]: We scan `Created` event from `TBTCSystem` contract so `depositCreated...` it's ok I think.

File: solidity/dashboard/src/services/token-staking.service.js (hunk 2 above)
Comment (r-czajkowski) [reply solidity/dashboard/src/services/token-staking.service.js:229]: yep, exactly. I'll try to add more color here in a moment

Comment (r-czajkowski) [reply solidity/dashboard/src/services/token-staking.service.js:229]: As a first step, we scan `StakeDelegated` by `owner` to get all operators. We can't 100% sure this relationship is still active since the owner can be changed.

Next, we need to scan `StakeOwnershipTransferred` by `newOwner` because another owner may have changed `owner` of a delegation to us. But again we can't 100% sure this relationship is still active since the owner can be changed.

So as the last step, we scan `StakeOwnershipTransferred` by all operators from these two results ^ and check the most recent `StakeOwnershipTransferred` event for a given operator to make sure we are the owner of the delegation.

**Review by pdyraga — APPROVED** [review]
